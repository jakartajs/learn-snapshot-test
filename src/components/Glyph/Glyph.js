import React, { PureComponent } from 'react';

import PropTypes from 'prop-types';
import classNaming from 'classnames';

import './Glyph.css';

class Glyph extends PureComponent {
  render() {
    const { className, name } = this.props;
    const stylename = classNaming(className, 'glyphicon', `glyphicon-${name}`);
    return <span className={stylename} />;
  }
}

Glyph.propTypes = {
  className: PropTypes.any,
  name: PropTypes.any,
};

Glyph.defaultProps = {
  className: '',
  name: '',
};

export default Glyph;
