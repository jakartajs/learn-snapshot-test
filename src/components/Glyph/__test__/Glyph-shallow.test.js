import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import React from 'react';
import { shallow } from 'enzyme';
import enzymeSerializer from 'enzyme-to-json/serializer';

import Glyph from '../Glyph';

expect.addSnapshotSerializer(enzymeSerializer);

jest.mock('classnames', () => {
  const data = (...props) => {
    return props[props.length-1];
  };
  return data;
});

describe('Glph Shallow Component', () => {
  test('should have empty span', () => {

    const result = shallow(<Glyph className={''} name={''} />);
    expect(result).toMatchSnapshot();
  });

  test('should have span with classname search', () => {
    const result = shallow(<Glyph className={''} name={'search'} />);
    expect(result).toMatchSnapshot();
  });
});
