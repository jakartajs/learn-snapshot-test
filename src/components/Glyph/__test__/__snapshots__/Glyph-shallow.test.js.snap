// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Glph Shallow Component should have empty span 1`] = `
<span
  className="glyphicon-"
/>
`;

exports[`Glph Shallow Component should have span with classname search 1`] = `
<span
  className="glyphicon-search"
/>
`;
